index.js

```diff
@@ -25,6 +24,15 @@ router.formio.mongoose = mongoose;
 // Use custom template delimiters.
 _.templateSettings.interpolate = /{{([\s\S]+?)}}/g;
 
+Promise.deferred = function() {
+  let result = {};
+  result.promise = new Promise(function(resolve, reject) {
+    result.resolve = resolve;
+    result.reject = reject;
+  });
+  return result;
+};
+
```

Comment:
We should avoid monkey patching built-in objects, especially global objects like Promise.